### PR TITLE
Convert unload test to json api format for `store.push`.. #3223

### PR DESCRIPTION
**Pull request by andrejunges**
This PR will break some tests, but now the the calls to store.push are
using jsonApiFormat

### CODE REVIEW

File: packages/ember-data/tests/integration/filter-test.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ var shouldNotContain = function(array, item) {
 
 module("integration/filter - DS.Model updating", {
   setup: function() {
-    array = [{ id: 1, name: "Scumbag Dale", bestFriend: 2 }, { id: 2, name: "Scumbag Katz" }, { id: 3, name: "Scumbag Bryn" }];
+    array = [{ "id": "1", "type": "person", "attributes": { "name": "Scumbag Dale", "bestFriend": "2" } }, { "id": "2", "type": "person", "attributes": { "name": "Scumbag Katz" } }, { "id": 3, "type": "person", "attributes": { "name": "Scumbag Bryn" } }];
```
Comment (wecc) [inline packages/ember-data/tests/integration/filter-test.js]: I think bestFriend should be a relationship here

**Comment by wecc** [comment]
Would you do squashing the commits and remove "WIP"?

File: packages/ember-data/tests/integration/filter-test.js [hunk 2]
```diff
@@ -16,8 +16,34 @@ var shouldNotContain = function(array, item) {
...
   setup: function() {
-    array = [{ id: 1, name: "Scumbag Dale", bestFriend: 2 }, { id: 2, name: "Scumbag Katz" }, { id: 3, name: "Scumbag Bryn" }];
-    Person = DS.Model.extend({ name: DS.attr('string'), bestFriend: DS.belongsTo('person', { inverse: null, async: false }) });
+    array = [{
+      "id": "1",
+      "type": "person",
+      "attributes": {
+        "name": "Scumbag Dale"
+      },
+      relationships: {
```
Comment (wecc) [inline packages/ember-data/tests/integration/filter-test.js]: Could you encapsulate all of the keys in quotes, for consistency?

File: packages/ember-data/tests/integration/filter-test.js [hunk 3]
```diff
@@ -16,8 +16,34 @@ var shouldNotContain = function(array, item) {
...
+        }
+      }
+    }, {
+      "id": "2",
+      "type": "person",
+      "attributes": {
+        "name": "Scumbag Katz"
+      }
+    }, {
+      "id": 3,
```
Comment (wecc) [inline packages/ember-data/tests/integration/filter-test.js]: Nitpicking here but as we want to conform to JSON-API as closely as possible IDs should be strings (http://jsonapi.org/format/#document-resource-object-identification).

There are a couple of places below too where IDs are numbers.

**Comment by wecc** [comment]
Can you prefix the commit with [CLEANUP beta]?

File: packages/ember-data/tests/integration/filter-test.js (hunk 2 above)
Comment (mixonic) [inline packages/ember-data/tests/integration/filter-test.js]: Perhaps the quoted keys should be unquoted instead @wecc?

File: packages/ember-data/tests/integration/filter-test.js (hunk 3 above)
Comment (mixonic) [inline packages/ember-data/tests/integration/filter-test.js]: Staying really close to JSON-API seems very important. There is great power in just telling people to check the spec when they have a question about formatting :clap:

**Comment by wecc** [comment]
Thank you so much!

**Comment by pangratz** [comment]
:+1: Awesome work @andrejunges!

**Pull request merged**